webapp/lib/datastore.dart

```diff
@@ -0,0 +1,68 @@
+
+class Dataset {
+  String name;
+  List<Message> messages;
+  List<Scheme> codeSchemes;
+
+  Dataset(this.name) {
+    messages = [];
+    codeSchemes = [];
+  }
+  Dataset.fromJson(Map jsonDataset) {
```

Comment:
Tbh, I haven't thought as to what's the convention, but if it takes a String here, then for consistency it would make sense to take a String in the other classes as well, which doesn't really work as I'm creating these objects hierarchically (one JSON -> Map conversion would be enough to parse the data).

Looking now at other Dart repos, the standard does indeed seem to be that a fromJson constructor takes a Map object rather than String, so I think it's good as it is?